File: Server-NodeJS/libs/hohoda/models/Conversation.js
import {Model} from "objection";

export default class Conversation extends Model {
    static get tableName() {
        return "conversation";
    }

    static get relationMappings() {
        return {
            person_one: {
                relation: Model.OneToOneRelation,
                modelClass: __dirname + '/Profile',
                join: {
                    from: 'conversation.member_one',
                    to: 'profile.user_id'
                }
            },
            person_two: {
                relation: Model.OneToOneRelation,
                modelClass: __dirname + '/Profile',
                join: {
                    from: 'conversation.member_two',
                    to: 'profile.user_id'
                }
            }
        }
    }
}

File: Client-Android/app/src/main/java/com/hodachop93/hohoda/model/Conversation.java
package com.hodachop93.hohoda.model;

import com.google.gson.annotations.SerializedName;

/**
 * Created by hopho on 04/05/2016.
 */
public class Conversation {
    private String id;
    @SerializedName("member_one")
    private String memberOne;
    @SerializedName("member_two")
    private String memberTwo;
    @SerializedName("person_one")
    private Profile personOne;
    @SerializedName("person_two")
    private Profile personTwo;
    @SerializedName("created_at")
    private long createdAt;
    @SerializedName("updated_at")
    private long updatedAt;
    @SerializedName("last_message")
    private String lastMessage;

    public String getId() {
        return id;
    }

    public String getMemberOne() {
        return memberOne;
    }

    public String getMemberTwo() {
        return memberTwo;
    }

    public Profile getPersonOne() {
        return personOne;
    }

    public Profile getPersonTwo() {
        return personTwo;
    }

    public long getCreatedAt() {
        return createdAt;
    }

    public long getUpdatedAt() {
        return updatedAt;
    }

    public String getLastMessage() {
        return lastMessage;
    }
}
